pass.go

```diff
@@ -0,0 +1,1054 @@
+}
+
+// NewHTTPError returns an instance of httpError which is required for
+// RenderErrorPage for proper display and error logging
+func NewHTTPError(code int, parentError error) error {
+	return httpError{code: code, error: parentError}
+}
+
+// ResolveTemplatePath returns the full path to the requested template file
+func (p *Pass) ResolveTemplatePath(templateSubPath string) (resolvedTemplatePath string) {
+	resolvedTemplatePath = filepath.Join(p.sv.cfg.TemplatePath, templateSubPath)
+
+	if _, err := os.Stat(resolvedTemplatePath); err != nil {
+		if os.IsNotExist(err) {
+			log.Fatalf("the template file %s was requested but it does not exist", resolvedTemplatePath)
+		} else {
+			log.Fatalf("the template file %s was requested but an unknown error occurred while loading it %v", resolvedTemplatePath, err)
+		}
+	}
+
+	return resolvedTemplatePath
+}
+
+func get(rv *requestVariables) (t *template.Template, tv *TemplateVariables, err error) {
+	tv = &TemplateVariables{}
+	if rv.r.Method == http.MethodGet {
+		t = rv.sv.templates.GetGet
+		tv.FormURL = ""
+
+		return t, tv, nil
+	}
+
+	t = rv.sv.templates.GetPost
+
+	var password string
+
+	entirePath := strings.Split(rv.r.URL.Path[1:], "/")
+	uuid := entirePath[0]
+
+	if password, err = rv.sv.db.get(uuid, rv.sv.cfg.Secret); err != nil {
+		return t, tv, err
+	}
+
+	if len(password) == 0 {
+		log.Printf("ip: %s requested a UUID but it was not found\n", rv.userIP)
+
+		tv.UUID = string(uuid)
+		return t, tv, passError{
+			error:    fmt.Errorf("UUID not found"),
+			location: "get()",
+		}
+	}
+
+	tv.Secret = string(password)
+
+	return t, tv, err
+}
+
+func create(rv *requestVariables) (*template.Template, *TemplateVariables, error) {
+	tv := &TemplateVariables{}
+	if rv.r.Method == http.MethodGet {
+		t := rv.sv.templates.CreateGet
+		tv.FormURL = "/"
+
+		return t, tv, nil
+	}
+
+	var uuid string
+	var err error
+	t := rv.sv.templates.CreatePost
+
+	var submittedPassword = rv.r.PostForm.Get("secret")
+
+	if len([]rune(submittedPassword)) < rv.sv.cfg.PasswordLength.Min ||
+		len([]rune(submittedPassword)) > rv.sv.cfg.PasswordLength.Max {
+		return t, tv, NewHTTPError(
+			http.StatusBadRequest,
+			fmt.Errorf("Password must be between %d and %d characters in length",
+				rv.sv.cfg.PasswordLength.Min,
+				rv.sv.cfg.PasswordLength.Max,
+			),
+		)
+	}
+
+	if uuid, err = rv.sv.db.create(submittedPassword, rv.sv.cfg.Secret); err != nil {
+		return t, tv, err
+	}
+
+	if len(uuid) == 0 {
+		log.Printf("Unable to create password entry in database!")
+		return t, tv, NewHTTPError(400, errors.New("An error occurred while creating password entry"))
+	}
+
+	tv.UUID = uuid
+	tv.URL = fmt.Sprintf("https://%s/%s", rv.sv.cfg.Hostname, string(uuid))
+
+	return t, tv, nil
+}
+
+// getIntegrityHash takes in a filename and returns a sha512 hash encoded as base64
+// used for Subresource Integrity (SRI)
+func (p *Pass) getIntegrityHash(filename string) (hash string) {
```

Comment:
`func (p *Pass) getIntegrityHash(filename string) string {`

Reply:
Fixed in next push